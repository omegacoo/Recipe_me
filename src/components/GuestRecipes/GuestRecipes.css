.GuestRecipes {
    text-align: center;
    margin-left: auto;
    margin-right: auto;
    box-shadow: 0 -5px 10px -5px;
    width: 100vw;
    height: 100vh;
}

.GuestRecipes_title {
    margin: 4vh;
    margin-bottom: 2vh;
    text-align: center;
    font-size: 10vw;
}

.GuestRecipesList_container {
    background-image: url('../../assets/open_book.png');
    background-size: 100% 100%;
    background-repeat: no-repeat;
    margin-left: 4%;
    margin-right: auto;
    display: block;
    width: 90vw;
    height: 55vh;
    padding: 2vh;
}

.GuestRecipesList {
    width: 80%;
    margin-left: 15%;
    margin-right: auto;
    margin-bottom: 1vh;
    height: 96%;
    overflow: auto;
    padding: 1vh;
    font-weight: bolder;
    font-size: 2vh;
    border-radius: 15px;
    background-color: #F0EAD6;
}

.GuestRecipes .ingredients {
    background-image: url('../../assets/boxed_food.png');
    background-color: transparent;
    background-repeat: no-repeat;
    background-size: 100%;
    width: 30vw;
    height: 18vh;
    border: none;
    display: block;
    margin-left: auto;
    margin-right: auto;
    font-weight: bolder;
}

.arrow {
    width: 50%;
    position: absolute;
    top: 25%;
    left: 5%;
}

/* iPad 768x1024 Portrait */
@media only screen and (min-width: 768px) and (min-height: 1024px) {
    .GuestRecipes_title {
        font-size: 8vw;
        margin-bottom: 0;
    }

    .GuestRecipesList_container {
        width: 80vw;
        margin-left: 8%;
    }

    .GuestRecipes .ingredients {
        width: 20vw;
    }

    .arrow {
        width: 50%;
        position: absolute;
        top: 20%;
        left: 5%;
    }
}

/* Desktop */
@media screen and (min-width: 1025px) {
    .GuestRecipes_title {
        font-size: 5vw;
    }

    .GuestRecipesList_container {
        width: 30vw;
        height: 50vh;
        margin-left: auto;
        margin-right: auto;
    }

    .GuestRecipes .ingredients {
        width: 8vw;
        height: 15vh;
        background-size: 100% 100%;
        margin-top: 2vh;
    }

    .GuestRecipes .ingredients:hover {
        cursor: pointer;
    }
    
    .GuestRecipes .ingredients:active {
        width: 7vw;
        height: 14vh;
        margin-top: 3vh;
    }

    .arrow {
        width: 20%;
        position: absolute;
        top: 20%;
        left: 30%;
    }    
}