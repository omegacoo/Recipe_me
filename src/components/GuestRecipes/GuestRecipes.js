import React from 'react';

import RecipeCard from '../RecipeCard/RecipeCard';

import arrow from '../../assets/arrow.png';

import './GuestRecipes.css';

import StoreContext from '../../StoreContext';

export default class GuestRecipes extends React.Component {
    static contextType = StoreContext;

    state = {
        guide: true
    };

    renderRecipes = () => {
        if(this.context.availableGuestRecipes.length > 0){
            return this.context.availableGuestRecipes.map(r =>
                <RecipeCard 
                    recipe={r}
                    key={r.id}
                />
            );
        } else {
            return 'Sorry, no recipes found...'
        };
    };

    handleIngredientsClick = () => {
        this.props.history.push('/guestpantry')
    };

    renderGuide = () => {
        if(this.state.guide){
            return <img src={arrow} alt='arrow' className='arrow pulse' />
        };
    };

    render(){
        return(
            <div className='Recipes'>
                <h1 className='Recipes_title'>&nbsp;Let's see what you can make!</h1>
                <section className='RecipesList_container'>
                    <ul className='RecipesList'>
                        {this.renderRecipes()}
                    </ul>
                </section>
                { this.renderGuide() }
                <button className='ingredients' onClick={this.handleIngredientsClick} />
            </div>
        );
    };
};