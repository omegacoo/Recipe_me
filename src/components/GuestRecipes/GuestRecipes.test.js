import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import GuestRecipes from './GuestRecipes';

describe('Recipes component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');

        ReactDOM.render(
            <BrowserRouter>
                <GuestRecipes />
            </BrowserRouter>, 
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders UI correctly', () => {
        expect(
        renderer.create(
            <BrowserRouter>
                <GuestRecipes />
            </BrowserRouter>).toJSON()
        ).toMatchSnapshot();
    });
});