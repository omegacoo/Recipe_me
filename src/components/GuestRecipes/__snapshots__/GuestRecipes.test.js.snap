// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipes component renders UI correctly 1`] = `
<div
  className="Recipes"
>
  <h1
    className="Recipes_title"
  >
    Â Let's see what you can make!
  </h1>
  <section
    className="RecipesList_container"
  >
    <ul
      className="RecipesList"
    >
      Sorry, no recipes found...
    </ul>
  </section>
  <img
    alt="arrow"
    className="arrow pulse"
    src="arrow.png"
  />
  <button
    className="ingredients"
    onClick={[Function]}
  />
</div>
`;
